src/fontra/views/editor/panel-reference-font.js

```diff
@@ -320,6 +322,38 @@ export default class ReferenceFontPanel extends Panel {
       await fontItem.fontFace.load();
     }
     this.model.referenceFontName = fontItem.fontIdentifier;
+
+    const font = new Font(fontItem.fontIdentifier);
+
+    font.onerror = (evt) => console.error(evt);
+    font.onload = (evt) => {
+      const font = evt.detail.font;
+      const getLangs = (table) => {
+        if (table) {
+          return table
+            .getSupportedScripts()
+            .reduce((acc, script) => {
+              const scriptTable = table.getScriptTable(script);
+              return acc.concat(table.getSupportedLangSys(scriptTable));
+            }, [])
+            .map((lang) => lang.trim());
+        } else {
+          return [];
+        }
+      };
+      const gsubLangs = getLangs(font.opentype.tables.GSUB);
+      const gposLangs = getLangs(font.opentype.tables.GPOS);
+      const allLangs = new Set([...gsubLangs, ...gposLangs]);
+      this.model.supportedLanguages = LANGUAGE_MAPPING.filter((l) =>
+        allLangs.has(l.ot)
+      );
+    };
+
+    const reader = new FileReader();
+    reader.addEventListener("load", () => {
+      font.fromDataBuffer(reader.result);
+    });
+    reader.readAsArrayBuffer(await readFontFileFromOPFS(fontItem.fontIdentifier));
```

Comment:
Have you tried to do it without FileReader, like this: `font.fromDataBuffer(await readFontFileFromOPFS(…))`

But either way, we need to do this at load time, so we don’t have to read the data twice.

Reply:
probably FileReader won't be needed if readFontFileFtomOPFS returns an array buffer. I should have missed that.

> But either way, we need to do this at load time, so we don’t have to read the data twice.

It must be loaded when a selected font change.

Reply:
> It must be loaded when a selected font change

Why would that be? We can do it at load, keep the result in a dict.

Reply:
I kept the selected languages in `fontItem` object with the others `uploadedFileName, fontIdentifier`. Is that ok?